### PR TITLE
Aha/dependency update

### PR DESCRIPTION
**Pull request by andyHa**

### CODE REVIEW

File: src/main/resources/default/templates/biz/tenants/tenant-details-fields.html.pasta [hunk 1]
```diff
@@ -72,7 +72,7 @@
                                         serviceUri="/tenants/autocomplete">
                 <i:if test="tenant.getParent().isFilled()">
                     <option value="@tenant.getParent().getId()" selected="selected">
-                        @tenant.getParent().getValue()
+                        @tenant.getParent().fetchValue()
```
Comment (meg1502) [inline src/main/resources/default/templates/biz/tenants/tenant-details-fields.html.pasta]: cache?

File: src/main/resources/default/templates/biz/tenants/select-user-account.html.pasta [hunk 2]
```diff
@@ -46,8 +46,8 @@
                     </i:if>
                 </td>
                 <td>
-                    @selectableUser.getTenant().getValue().as(sirius.biz.tenants.Tenant.class).getTenantData().getName()<br>
-                    @selectableUser.getTenant().getValue().as(sirius.biz.tenants.Tenant.class).getTenantData().getAccountNumber()
+                    @selectableUser.getTenant().fetchValue().as(sirius.biz.tenants.Tenant.class).getTenantData().getName()<br>
```
Comment (meg1502) [inline src/main/resources/default/templates/biz/tenants/select-user-account.html.pasta]: cache?

File: src/main/resources/default/templates/biz/tenants/select-user-account.html.pasta [hunk 3]
```diff
@@ -46,8 +46,8 @@
...
+                    @selectableUser.getTenant().fetchValue().as(sirius.biz.tenants.Tenant.class).getTenantData().getAccountNumber()
```
Comment (meg1502) [inline src/main/resources/default/templates/biz/tenants/select-user-account.html.pasta]: cache?

File: src/main/java/sirius/biz/tenants/mongo/MongoUserAccount.java [hunk 4]
```diff
@@ -52,8 +52,8 @@ protected void enhanceSearchField() {
         addContent(getUserAccountData().getLogin().getUsername());
         addContent(getUserAccountData().getEmail());
 
-        addContent(getTenant().getValue().getTenantData().getName());
-        addContent(getTenant().getValue().getTenantData().getAccountNumber());
+        addContent(getTenant().fetchValue().getTenantData().getName());
```
Comment (meg1502) [inline src/main/java/sirius/biz/tenants/mongo/MongoUserAccount.java]: cache?

File: src/main/java/sirius/biz/tenants/mongo/MongoUserAccount.java [hunk 5]
```diff
@@ -52,8 +52,8 @@ protected void enhanceSearchField() {
...
+        addContent(getTenant().fetchValue().getTenantData().getAccountNumber());
```
Comment (meg1502) [inline src/main/java/sirius/biz/tenants/mongo/MongoUserAccount.java]: cache?

File: src/main/java/sirius/biz/tenants/UserAccountData.java [hunk 6]
```diff
@@ -176,7 +176,7 @@ public int getSanePasswordLength() {
     }
 
     protected Tenant<?> getTenant() {
-        return ((UserAccount<?, ?>) userObject).getTenant().getValue();
+        return ((UserAccount<?, ?>) userObject).getTenant().fetchValue();
```
Comment (meg1502) [inline src/main/java/sirius/biz/tenants/UserAccountData.java]: cache?

File: src/main/java/sirius/biz/tenants/UserAccountController.java [hunk 7]
```diff
@@ -375,8 +375,8 @@ public void forgotPassword(final WebContext ctx, JSONStructuredOutput out) {
         auditLog.neutral("AuditLog.resetPassword")
                 .causedByUser(account.getUniqueName(), account.getUserAccountData().getLogin().getUsername())
                 .forUser(account.getUniqueName(), account.getUserAccountData().getLogin().getUsername())
-                .forTenant(account.getTenant().getValue().getIdAsString(),
-                           account.getTenant().getValue().getTenantData().getName())
+                .forTenant(account.getTenant().fetchValue().getIdAsString(),
+                           account.getTenant().fetchValue().getTenantData().getName())
```
Comment (meg1502) [inline src/main/java/sirius/biz/tenants/UserAccountController.java]: cache?

File: src/main/java/sirius/biz/tenants/TenantUserManager.java [hunk 8]
```diff
@@ -717,20 +717,20 @@ protected U getUserObject(UserInfo userInfo) {
     protected UserSettings getUserSettings(UserSettings scopeSettings, UserInfo userInfo) {
         U user = userInfo.getUserObject(getUserClass());
         if (user.getUserAccountData().getPermissions().getConfig() == null) {
-            if (user.getTenant().getValue().getTenantData().getConfig() == null) {
+            if (user.getTenant().fetchValue().getTenantData().getConfig() == null) {
```
Comment (meg1502) [inline src/main/java/sirius/biz/tenants/TenantUserManager.java]: cache?

File: src/main/java/sirius/biz/tenants/SAMLController.java [hunk 9]
```diff
@@ -200,7 +200,7 @@ private T findTenant(SAMLResponse response) {
 
     private void verifyUser(SAMLResponse response, UserInfo user) {
         U account = user.getUserObject(getUserClass());
-        T tenant = account.getTenant().getValue();
+        T tenant = account.getTenant().fetchValue();
```
[the remark this answers is not in the document]
Comment (andyHa) [reply src/main/java/sirius/biz/tenants/SAMLController.java]: not required here - a user object and all related entities reside already in a cache

File: src/main/java/sirius/biz/tenants/TenantController.java [hunk 10]
```diff
@@ -354,7 +354,7 @@ public void selectTenant(final WebContext ctx, String id) {
                     .causedByUser(account.getUniqueName(), account.getUserAccountData().getLogin().getUsername())
                     .forUser(account.getUniqueName(), account.getUserAccountData().getLogin().getUsername())
                     .forTenant(account.getTenant().getIdAsString(),
-                               account.getTenant().getValue().getTenantData().getName())
+                               account.getTenant().fetchValue().getTenantData().getName())
```
[the remark this answers is not in the document]
Comment (andyHa) [reply src/main/java/sirius/biz/tenants/TenantController.java]: also served by the user manage from cache

File: src/main/java/sirius/biz/tenants/TenantUserManager.java [hunk 11]
```diff
@@ -468,7 +468,7 @@ private UserInfo verifyIpRange(WebContext ctx, UserInfo info) {
             return defaultUser;
         }
 
-        T tenant = account.getTenant().getValue();
+        T tenant = account.getTenant().fetchValue();
```
[the remark this answers is not in the document]
Comment (andyHa) [reply src/main/java/sirius/biz/tenants/TenantUserManager.java]: fetchAccount does that already

File: src/main/java/sirius/biz/tenants/TenantUserManager.java [hunk 12]
```diff
@@ -547,7 +547,7 @@ private UserInfo asUserWithRoles(U account, Set<String> roles, @Nullable Supplie
         return UserInfo.Builder.createUser(account.getUniqueName())
                                .withUsername(account.getUserAccountData().getLogin().getUsername())
                                .withTenantId(String.valueOf(account.getTenant().getId()))
-                               .withTenantName(account.getTenant().getValue().getTenantData().getName())
+                               .withTenantName(account.getTenant().fetchValue().getTenantData().getName())
```
[the remark this answers is not in the document]
Comment (andyHa) [reply src/main/java/sirius/biz/tenants/TenantUserManager.java]: s.a.

File: src/main/java/sirius/biz/tenants/TenantUserManager.java [hunk 13]
```diff
@@ -600,7 +600,7 @@ public UserInfo findUserByCredentials(@Nullable WebContext ctx, String user, Str
         auditLog.negative("AuditLog.loginRejected")
                 .forUser(account.getUniqueName(), account.getUserAccountData().getLogin().getUsername())
                 .forTenant(String.valueOf(account.getTenant().getId()),
-                           account.getTenant().getValue().getTenantData().getName())
+                           account.getTenant().fetchValue().getTenantData().getName())
```
[the remark this answers is not in the document]
Comment (andyHa) [reply src/main/java/sirius/biz/tenants/TenantUserManager.java]: s.a.

File: src/main/java/sirius/biz/tenants/TenantUserManager.java [hunk 14]
```diff
@@ -644,7 +644,7 @@ protected void completeAuditLogForUser(AuditLog.AuditLogBuilder builder, U accou
         builder.causedByUser(account.getUniqueName(), account.getUserAccountData().getLogin().getUsername())
                .forUser(account.getUniqueName(), account.getUserAccountData().getLogin().getUsername())
                .forTenant(String.valueOf(account.getTenant().getId()),
-                          account.getTenant().getValue().getTenantData().getName())
+                          account.getTenant().fetchValue().getTenantData().getName())
```
[the remark this answers is not in the document]
Comment (andyHa) [reply src/main/java/sirius/biz/tenants/TenantUserManager.java]: s.a.

File: src/main/java/sirius/biz/tenants/TenantUserManager.java (hunk 8 above)
Comment (andyHa) [reply src/main/java/sirius/biz/tenants/TenantUserManager.java]: s.a.

File: src/main/java/sirius/biz/tenants/TenantUserManager.java [hunk 15]
```diff
@@ -717,20 +717,20 @@ protected U getUserObject(UserInfo userInfo) {
...
         if (user.getUserAccountData().getPermissions().getConfig() == null) {
-            if (user.getTenant().getValue().getTenantData().getConfig() == null) {
+            if (user.getTenant().fetchValue().getTenantData().getConfig() == null) {
                 return scopeSettings;
             }
 
             return configCache.get(user.getTenant().getUniqueObjectName(), i -> {
                 Config cfg = scopeSettings.getConfig();
-                cfg = user.getTenant().getValue().getTenantData().getConfig().withFallback(cfg);
+                cfg = user.getTenant().fetchValue().getTenantData().getConfig().withFallback(cfg);
```
[the remark this answers is not in the document]
Comment (andyHa) [reply src/main/java/sirius/biz/tenants/TenantUserManager.java]: s.a.

File: src/main/java/sirius/biz/tenants/TenantUserManager.java [hunk 16]
```diff
@@ -717,20 +717,20 @@ protected U getUserObject(UserInfo userInfo) {
...
                 return Tuple.create(new UserSettings(cfg), user.getTenant().getUniqueObjectName());
             }).getFirst();
         }
 
         return configCache.get(user.getUniqueName(), i -> {
             Config cfg = scopeSettings.getConfig();
-            cfg = user.getTenant().getValue().getTenantData().getConfig().withFallback(cfg);
+            cfg = user.getTenant().fetchValue().getTenantData().getConfig().withFallback(cfg);
```
[the remark this answers is not in the document]
Comment (andyHa) [reply src/main/java/sirius/biz/tenants/TenantUserManager.java]: s.a.

File: src/main/java/sirius/biz/tenants/TenantUserManager.java [hunk 17]
```diff
@@ -849,7 +849,7 @@ private boolean isWithinInterval(LocalDateTime dateTime, Integer requiredInterva
         }
 
         Set<String> roles = computeRoles(user,
-                                         user.getTenant().getValue(),
+                                         user.getTenant().fetchValue(),
```
[the remark this answers is not in the document]
Comment (andyHa) [reply src/main/java/sirius/biz/tenants/TenantUserManager.java]: s.a.

File: src/main/java/sirius/biz/tenants/TenantUserManager.java [hunk 18]
```diff
@@ -926,7 +926,7 @@ protected String computeLang(WebContext ctx, String userId) {
             return NLS.getDefaultLanguage();
         }
         return Strings.firstFilled(userAccount.getUserAccountData().getLang(),
-                                   userAccount.getTenant().getValue().getTenantData().getLang(),
+                                   userAccount.getTenant().fetchValue().getTenantData().getLang(),
```
[the remark this answers is not in the document]
Comment (andyHa) [reply src/main/java/sirius/biz/tenants/TenantUserManager.java]: s.a.

File: src/main/java/sirius/biz/tenants/Tenants.java [hunk 19]
```diff
@@ -114,7 +114,7 @@ public boolean hasUser() {
      */
     @Nonnull
     public Optional<T> getCurrentTenant() {
-        return getCurrentUser().flatMap(u -> Optional.ofNullable(u.getTenant().getValue()));
+        return getCurrentUser().flatMap(u -> Optional.ofNullable(u.getTenant().fetchValue()));
```
[the remark this answers is not in the document]
Comment (andyHa) [reply src/main/java/sirius/biz/tenants/Tenants.java]: the curret user will always have its tenant initialized by the TenantUserManager

File: src/main/java/sirius/biz/tenants/UserAccountData.java (hunk 6 above)
Comment (andyHa) [reply src/main/java/sirius/biz/tenants/UserAccountData.java]: pre-filled

**Pull request merged**